Packs/Jira/Integrations/JiraV2/JiraV2.py

```diff
@@ -552,11 +552,34 @@ def get_project_id(project_key='', project_name=''):
     if not project_key and not project_name:
         return_error('You must provide at least one of the following: project_key or project_name')
 
-    result = jira_req('GET', 'rest/api/latest/issue/createmeta', resp_type='json')
+    result: dict = {}
+
+    try:
+        result = jira_req('GET', 'rest/api/latest/issue/createmeta', resp_type='json')
+    except DemistoException as de:
+
+        if de.message != 'Status code: 404\nMessage: Issue Does Not Exist':
+            raise de
+
+        demisto.debug(f'Could not find expected Jira endpoint: {BASE_URL}/api/latest/issue/createmeta.'
+                      f'Trying another endpoint: {BASE_URL}/api/latest/project.')
+
+        # a new endpoint for Jira version 9.0.0 and above, so we execute another api call
+        result = jira_req('GET', 'rest/api/latest/project', resp_type='json')
+
+        # Jira's response changed to a list of projects from version 9.0.0
+        projects_lst = list(filter(lambda x: x.get('key').lower() == project_key.lower() or x.get('name').lower() == project_name.lower(), result))
+
+        # Filtering should give us a list with one project, only one project should match the filter's conditions
+        if projects_lst:
+            return projects_lst[0].get('id')
+
+    # Jira used to respond with a dictionary with the 'projects' key until version 9.0.0
+    if isinstance(result, dict):
+        for project in result.get('projects', []):
```

Comment:
project should be a dict, so the default value (though empty) should also be a dict.